TF Lite (CMake): neon2sse missing license file location

<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

2.9.0

### Custom Code

No

### OS Platform and Distribution

Linux

### Mobile device

_No response_

### Python version

3.8

### Bazel version

N/A

### GCC/Compiler version

N/A

### CUDA/cuDNN version

N/A

### GPU model and memory

N/A

### Current Behaviour?

```shell
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake tracks licenses for each dependency of an imported module of TensorFlow Lite. This allows projects using TF Lite to build an aggregate license file for redistribution (e.g https://github.com/google-research/falken/blob/e15aea40aec08e1e5858905e7dd6de7da2e4a1ec/sdk/cpp/CMakeLists.txt#L196 )

https://github.com/tensorflow/tensorflow/commit/1485091a6a64989a01cd2db76d53fe71653ae666 caused a regression in TF Lite's CMake dependencies by manually specifying a download URL where the license URL can no longer be inferred by `OverridableFetchContent.cmake`.

I'm happy to provide a fix that:
* Specifies the license URL and downloaded location.
* Enables license file checking in the TF Lite CMake project.
```


### Standalone code to reproduce the issue

```shell
Add `set(OVERRIDABLE_FETCH_CONTENT_LICENSE_CHECK ON)` near the top of `tensorflow/lite/CMakeLists.txt`, configure the project and wait for configuration to fail due to neon2sse missing a license.
```


### Relevant log output

```shell
$ cd tensorflow/lite
$ mkdir build
$ cd build
$ cmake ..
# lots of output
- Could NOT find CLANG_FORMAT: Found unsuitable version "0.0", but required is exact version "9" (found CLANG_FORMAT_EXECUTABLE-NOTFOUND)
--
-- Configured Eigen 3.4.90
--
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT
-- Performing Test FARMHASH_HAS_BUILTIN_EXPECT - Success
-- Looking for strtof_l
-- Looking for strtof_l - found
-- Looking for strtoull_l
-- Looking for strtoull_l - found
CMake Error at tools/cmake/modules/OverridableFetchContent.cmake:571 (message):
  Required license URL not found for neon2sse
Call Stack (most recent call first):
  tools/cmake/modules/neon2sse.cmake:37 (OverridableFetchContent_Populate)
  tools/cmake/modules/Findneon2sse.cmake:18 (include)
  CMakeLists.txt:148 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/stewart/src/tensorflow/tensorflow/lite/build/CMakeFiles/CMakeOutput.log".
See also "/home/stewart/src/tensorflow/tensorflow/lite/build/CMakeFiles/CMakeError.log".
```
</details>

--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -138,6 +138,8 @@ macro(populate_tf_source_vars RELATIVE_DIR SOURCES_VAR)
     "${TF_SOURCE_DIR}/${RELATIVE_DIR}" ${SOURCES_VAR} ${ARGN}
   )
 endmacro()
+# Make sure all repos have licenses.
+set(OVERRIDABLE_FETCH_CONTENT_LICENSE_CHECK ON)
 # Find TensorFlow Lite dependencies.
 find_package(absl REQUIRED)
 find_package(Eigen3 REQUIRED)

--- a/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
+++ b/tensorflow/lite/tools/cmake/modules/OverridableFetchContent.cmake
@@ -431,7 +431,10 @@ function(_OverridableFetchContent_SetProperty CONTENT_NAME PROPERTY_NAME
     BRIEF_DOCS "${DOCUMENTATION}"
     FULL_DOCS "${DOCUMENTATION}"
   )
-  set_property(GLOBAL PROPERTY "${GLOBAL_PROPERTY_NAME}" "${PROPERTY_VALUE}")
+set_property(
+  GLOBAL PROPERTY "${GLOBAL_PROPERTY_NAME}"
+  "${PROPERTY_VALUE}"
+)
 endfunction()
 
 
@@ -505,7 +508,10 @@ function(OverridableFetchContent_GetProperties CONTENT_NAME)
     )
     set(PROPERTY_VALUE "${${EXPORT_PROPERTY}}")
     if(PROPERTY_VALUE)
-      set(${CONTENT_NAME}_${EXPORT_PROPERTY} "${PROPERTY_VALUE}" PARENT_SCOPE)
+      set(${CONTENT_NAME_LOWER}_${EXPORT_PROPERTY}
+        "${PROPERTY_VALUE}"
+        PARENT_SCOPE
+      )
     endif()
   endforeach()
   FetchContent_GetProperties("${CONTENT_NAME}" ${OUTPUT_ARGS})
@@ -534,8 +540,8 @@ function(OverridableFetchContent_Populate CONTENT_NAME)
   # If a license file isn't cached try finding it in the repo.
   set(LICENSE_FILE "${${CONTENT_NAME_LOWER}_LICENSE_FILE}")
   set(LICENSE_URL "${${CONTENT_NAME_LOWER}_LICENSE_URL}")
+  set(SOURCE_DIR "${${CONTENT_NAME_LOWER}_SOURCE_DIR}")
   if(${CONTENT_NAME}_POPULATED AND NOT LICENSE_FILE)
-    set(SOURCE_DIR "${${CONTENT_NAME_LOWER}_SOURCE_DIR}")
     find_file(_${CONTENT_NAME_LOWER}_LICENSE_FILE
       NAMES LICENSE LICENSE.md LICENSE.txt NOTICE COPYING
       PATHS "${SOURCE_DIR}"
@@ -548,6 +554,33 @@ function(OverridableFetchContent_Populate CONTENT_NAME)
       file(RELATIVE_PATH LICENSE_FILE "${SOURCE_DIR}" "${LICENSE_FILE}")
       file(TO_CMAKE_PATH "${LICENSE_FILE}" LICENSE_FILE)
     endif()
+  endif()
+  # If LICENSE_FILE was not found but a URL was specified then try downloading
+  # the license.
+  set(LICENSE_FILE_FULL_PATH "${SOURCE_DIR}/${LICENSE_FILE}")
+  if(NOT EXISTS "${LICENSE_FILE_FULL_PATH}" AND LICENSE_URL)
+    set(LICENSE_FILE_DOWNLOAD "${SOURCE_DIR}/${CONTENT_NAME}_LICENSE.txt")
+    if(NOT EXISTS "${LICENSE_FILE_DOWNLOAD}")
+      message(STATUS
+        "${CONTENT_NAME} '${LICENSE_FILE_FULL_PATH}' does not exist "
+        "downloading ${LICENSE_URL} --> ${LICENSE_FILE_DOWNLOAD}")
+      file(DOWNLOAD "${LICENSE_URL}" "${LICENSE_FILE_DOWNLOAD}" STATUS RESULT)
+      list(GET RESULT 0 RESULT)
+      if(NOT RESULT EQUAL 0)
+        message(
+          FATAL_ERROR
+          "Failed to download ${LICENSE_URL} for ${CONTENT_NAME} to "
+          "${LICENSE_FILE_DOWNLOAD}"
+        )
+      endif()
+    endif()
+    file(RELATIVE_PATH LICENSE_FILE "${SOURCE_DIR}" "${LICENSE_FILE_DOWNLOAD}")
+    _OverridableFetchContent_SetProperty(
+      "${CONTENT_NAME}"
+      LICENSE_FILE
+      "License for ${CONTENT_NAME}"
+      "${LICENSE_FILE}"
+    )
   endif()
   # If a LICENSE_FILE was found populate the URL.
   if(LICENSE_FILE AND NOT LICENSE_URL)

--- a/tensorflow/lite/tools/cmake/modules/egl_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/egl_headers.cmake
@@ -28,6 +28,8 @@ OverridableFetchContent_Declare(
   GIT_PROGRESS TRUE
   PREFIX "${CMAKE_BINARY_DIR}"
   SOURCE_DIR "${CMAKE_BINARY_DIR}/egl_headers"
+  # Per https://www.khronos.org/legal/Khronos_Apache_2.0_CLA
+  LICENSE_URL "https://www.apache.org/licenses/LICENSE-2.0.txt"
 )
 
 OverridableFetchContent_GetProperties(egl_headers)

--- a/tensorflow/lite/tools/cmake/modules/neon2sse.cmake
+++ b/tensorflow/lite/tools/cmake/modules/neon2sse.cmake
@@ -19,11 +19,16 @@ if(TARGET neon2sse OR neon2sse_POPULATED)
   return()
 endif()
 
+set(NEON2SSE_URL
+  https://storage.googleapis.com/mirror.tensorflow.org/github.com/intel/ARM_NEON_2_x86_SSE/archive/a15b489e1222b2087007546b4912e21293ea86ff.tar.gz
+)
 OverridableFetchContent_Declare(
   neon2sse
-  URL https://storage.googleapis.com/mirror.tensorflow.org/github.com/intel/ARM_NEON_2_x86_SSE/archive/a15b489e1222b2087007546b4912e21293ea86ff.tar.gz
+  URL "${NEON2SSE_URL}"
   # Sync with tensorflow/workspace2.bzl
   URL_HASH SHA256=019fbc7ec25860070a1d90e12686fc160cfb33e22aa063c80f52b363f1361e9d
+  LICENSE_FILE LICENSE
+  LICENSE_URL "${NEON2SSE_URL}"
   SOURCE_DIR "${CMAKE_BINARY_DIR}/neon2sse"
 )
 

--- a/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake
+++ b/tensorflow/lite/tools/cmake/modules/opengl_headers.cmake
@@ -28,6 +28,8 @@ OverridableFetchContent_Declare(
   GIT_PROGRESS TRUE
   PREFIX "${CMAKE_BINARY_DIR}"
   SOURCE_DIR "${CMAKE_BINARY_DIR}/opengl_headers"
+  # Per https://www.khronos.org/legal/Khronos_Apache_2.0_CLA
+  LICENSE_URL "https://www.apache.org/licenses/LICENSE-2.0.txt"
 )
 
 OverridableFetchContent_GetProperties(opengl_headers)

